packages/cozy-harvest-lib/src/components/DeleteAccountButton.jsx

```diff
@@ -22,16 +22,18 @@ export class DeleteAccountButton extends Component {
     const { account, deleteAccount, onError, onSuccess } = this.props
     try {
       await deleteAccount(account)
-      if (typeof onSuccess === 'function') onSuccess(account)
+      this.setState({ deleting: false }, () => {
+        if (typeof onSuccess === 'function') onSuccess(account)
+      })
     } catch (e) {
-      if (typeof onError === 'function') {
-        onError(e)
-      } else {
-        // eslint-disable-next-line no-console
-        console.error(e)
-      }
-    } finally {
-      this.setState({ deleting: false })
+      this.setState({ deleting: false }, () => {
+        if (typeof onError === 'function') {
+          onError(e)
+        } else {
+          // eslint-disable-next-line no-console
+          console.error(e)
+        }
```

Comment:
In this case, why not putting `this.setState({deleting: false})` before the `try` block ?

Reply:
Because we have to wait for the `deleteAccount` be finished before changing this state

Reply:
I would prefer to keep the finally block and having a cancelable promise during the unmount to not set the state. Or having a `this._isMounted` in the finally block. WDYT ?

Reply:
FMC, what is the issue to rely on something provided by the `setState` API here?

Reply:
Nothing  really wrong, but as we want to setState({deleting: false}) in each case, we should deal with it only once. 

Also, do we really considered that we have managed the `deletion` before calling the success or the `onError` callback? Not sure since they are part of the worklow. 

I would avoid to trigger a render before calling the callback. 

This is why I prefer to put the setState in the finally block and test if the component is still mounted.